### DISCUSSION TITLE
automation tokens cannot perform `npm access ls-packages`

### DISCUSSION THREAD

**Discussion by ThisIsMissEm** [npm/feedback#698] [Registry] [answered]
In lerna, by default, it attempts to verify that you can actually publish the packages it will publish, by using `npm access ls-packages` (well, the API behind it). It seems that automation tokens don't allow that action, which breaks lerna, unless you use `--no-verify-access`

It's also unclear as to whether automation tokens can interact with `npm dist-tag` as to remove tags that are published just for CI/CD (we're working on moving away from doing this)

**Comment by monishcm** [comment, marked as answer]
Automation tokens are primarily used for publishing, whoami and few other APIs. The said API is not supported by automation tokens. That said, we are making some improvements to make Tokens more usable and granular.

**Reply by ThisIsMissEm** [reply]
Is there perhaps an alternative API that lerna could use to check if it could publish, if automation tokens cannot issue a `npm access ls-packages` request? Or would each package have to manually be checked somehow?

Lerna does the check to see if you can publish to the packages in the mono-repo as to not end up with half published releases due to permission errors.
